src/amoc_controller.erl

```diff
@@ -305,89 +246,21 @@ start_scenario(Scenario, UserIds, State) ->
                [Start, End, Length]),
     start_users(Scenario, UserIds, interarrival(), State).
 
--spec init_scenario(amoc:scenario()) -> any().
-init_scenario(Scenario) ->
-    case erlang:function_exported(Scenario, init, 0) of
-        true ->
-            case Scenario:init() of
-                ok -> {ok, ok};
-                RetValue -> RetValue
+-spec init_scenario(amoc:scenario(), amoc:do_opts()) -> any().
+init_scenario(Scenario, Opts) ->
+    ScenarioConfig = proplists:get_value(config, Opts, []),
+    case amoc_config_scenario:parse_scenario_settings(Scenario, ScenarioConfig) of
+        ok ->
+            case erlang:function_exported(Scenario, init, 0) of
```

Comment:
I'd like to see this case extracted to a separate function named `call_scenario_init` or sth like that. The reason why I want this is that currently have 3 levels of cases in single function.

Reply:
amoc_controller will be heavily reworked in the next PR. and this code won't be there any more